app/controllers/application_controller.rb

```diff
@@ -501,6 +513,10 @@ def secure_session
 
   private
 
+  def content_security_policy
+    @content_security_policy ||= ContentSecurityPolicy.new.build
```

Comment:
Do we need to cache this so we don't have to calculate it for every request? Note the cache needs to be invalidated when the related site settings update